### PR TITLE
Add contributing icons docs

### PR DESCRIPTION
**Pull request by robinmetral**
## Purpose

This adds documentation for contributing icons to `@sumup/icons`.

## Approach and changes

- document the process of adding icons as well as caveats
  - note that these docs include internal links. This is acceptable here because icons are meant to be added by internal contributors with access to the Figma icons library.
- delete a couple of unused assets

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* ~Unit and integration tests~
* ~Meets minimum browser support~
* ~Meets accessibility requirements~

### CODE REVIEW

File: docs/contributing/contributing-icons.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+# Contributing Icons
+
+This page outlines the process of contributing an icon to the `@sumup/icons` package.
+
+> Note that icons should be added by internal contributors with access to the SumUp [Figma icons library](https://www.figma.com/file/vnFVuPNlqF45rkw1u9toBC/SumUp-Iconography) (internal link). If you don't have access but would like to see an icon added to `@sumup/icons`, please [open an issue](https://github.com/sumup-oss/circuit-ui/issues/new).
+
+## Adding a new icon to `@sumup/icons`
+
+1. Create a new SVG file for each icon size in [`packages/web/icons/v2/`](https://github.com/sumup-oss/circuit-ui/tree/main/packages/icons/web/v2) with the name `name_size.svg` (e.g. `add_items_24`—this will generate an `<AddItems />` component).
+2. Export the icon as SVG from the [Figma icons library](https://www.figma.com/file/vnFVuPNlqF45rkw1u9toBC/SumUp-Iconography) (internal link). If the icon isn't in the library, make a request with the design team first.
```
Comment (connor-baer) [inline docs/contributing/contributing-icons.stories.mdx:14]: Huh, I always used the `Export` panel in the right sidebar. This is easier!

File: docs/contributing/contributing-icons.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+   ![Right click on the group in Figma and choose "export as SVG"](/images/contributing-icons-export.png)
+3. Paste the SVG into your file and verify the code—refer to the "Caveats" section below.
```
Comment (connor-baer) [inline docs/contributing/contributing-icons.stories.mdx:16]: 🙈
```suggestion
3. Paste the SVG into your file and verify the code — refer to the "Caveats" section below.
```

File: docs/contributing/contributing-icons.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+4. Commit the icons, they will automatically be optimized in the commit hook using `svgo`.
```
Comment (connor-baer) [inline docs/contributing/contributing-icons.stories.mdx:17]:
```suggestion
4. Commit the icons, they will automatically be optimized in the precommit hook using `svgo`.
```

File: docs/contributing/contributing-icons.stories.mdx (hunk 1 above)
Comment (robinmetral) [reply docs/contributing/contributing-icons.stories.mdx:14]: Yes, I also find it easier than having to find the file in your downloads and move/rename it 😅 Personal preference but that's a way

File: docs/contributing/contributing-icons.stories.mdx (hunk 2 above)
Comment (robinmetral) [reply docs/contributing/contributing-icons.stories.mdx]: Actually, from [a Merriam Webster article](https://www.merriam-webster.com/words-at-play/em-dash-en-dash-how-to-use):

> Spacing around an em dash varies. Most newspapers insert a space before and after the dash, and many popular magazines do the same, but most books and journals omit spacing, closing whatever comes before and after the em dash right up next to it. This website prefers the latter, its style requiring the closely held em dash in running text.

...but I don't mind, let's be newspapery and keep the spaces 😛 learned something about punctuation this morning, usually used without spaces (personal pref and probably what I'm used to from books)

Comment (connor-baer) [reply docs/contributing/contributing-icons.stories.mdx]: TIL! I find it easier to read with the spaces around but 🤷‍♂️

**Review by connor-baer — APPROVED** [review]

**Pull request merged**